react/components/ProductMessagesExport.tsx

```diff
@@ -88,24 +88,27 @@ const ProductMessagesExport: FC<Props> = ({
   showToast,
   stepCounterControl,
 }) => {
-  const [exporting, setExporting] = useState(false)
-  const [exported, setExported] = useState(false)
+  const [state, setState] = useState<State>(null)
```

Comment:
`setState` sounds too generic and resembles the old React state functions.

I get it you are creating a very simple state machine here. Can you think of a more specific name that better express what kind of state is it holding?

Reply:
Do you think `exportState` is good enough?